### PR TITLE
Updated "liking", added "favoriting" and "completing"

### PR DESCRIPTION
**Pull request by jackhorton**

### CODE REVIEW

File: src/controllers/completed.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        action: 'made',
+        value: false
+    }).then(() => {
+        res.status(200).send({
+            status: 'success'
+        });
+    }).catch(() => {
+        res.status(403).send({
+            status: 'failure',
+            error: 'Can not favorite this recipe'
```
Comment (garrickb) [inline src/controllers/completed.js]: what

File: src/controllers/completed.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        });
+
+        next(new Error(`User ${req.user.id} can not delete favorited recipe ${req.body.recipe_id}`));
```
Comment (garrickb) [inline src/controllers/completed.js]: what

File: src/controllers/completed.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        res.status(200).send({
+            status: 'success'
+        });
+    }).catch(() => {
+        res.status(403).send({
+            status: 'failure',
+            error: 'Can not favorite this recipe'
+        });
+
+        next(new Error(`User ${req.user.id} can not favorite recipe ${req.body.recipe_id}`));
```
Comment (garrickb) [inline src/controllers/completed.js]: what

File: src/controllers/favorites.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        action: 'favorited',
+        value: false
+    }).then(() => {
+        res.status(200).send({
+            status: 'success'
+        });
+    }).catch(() => {
+        res.status(403).send({
+            status: 'failure',
+            error: 'Can not favorite this recipe'
```
Comment (garrickb) [inline src/controllers/favorites.js]: what

File: src/controllers/favorites.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        }));
+    }).catch(() => {
+        res.status(403).send({
+            status: 'failure',
+            error: 'Unable to retrieve favorited recipes'
+        });
+
+        next(new Error(`Unable to load favorited recipes for user ${req.user.id}`));
+    });
+});
```
Comment (garrickb) [inline src/controllers/favorites.js:48]: Should return the total number of results.

File: src/controllers/favorites.js (hunk 4 above)
Comment (jackhorton) [inline src/controllers/favorites.js]: why is this a what

Comment (garrickb) [inline src/controllers/favorites.js]: It's doing the opposite of favoriting

**Pull request merged**